src/python/pants/backend/codegen/thrift/dependency_inference.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import re
+from collections import defaultdict
+from dataclasses import dataclass
+from typing import DefaultDict
+
+from pants.backend.codegen.thrift.subsystem import ThriftSubsystem
+from pants.backend.codegen.thrift.target_types import AllThriftTargets, ThriftSourceField
+from pants.core.util_rules.stripped_source_files import StrippedSourceFileNames
+from pants.engine.addresses import Address
+from pants.engine.fs import Digest, DigestContents
+from pants.engine.rules import Get, MultiGet, collect_rules, rule
+from pants.engine.target import (
+    Dependencies,
+    DependenciesRequest,
+    ExplicitlyProvidedDependencies,
+    HydratedSources,
+    HydrateSourcesRequest,
+    InferDependenciesRequest,
+    InferredDependencies,
+    SourcesPathsRequest,
+    WrappedTarget,
+)
+from pants.engine.unions import UnionRule
+from pants.util.frozendict import FrozenDict
+from pants.util.logging import LogLevel
+from pants.util.ordered_set import FrozenOrderedSet, OrderedSet
+
+
+@dataclass(frozen=True)
+class ThriftMapping:
+    """A mapping of stripped .thrift file names to their owning file address."""
+
+    mapping: FrozenDict[str, Address]
+    ambiguous_modules: FrozenDict[str, tuple[Address, ...]]
+
+
+@rule(desc="Creating map of Thrift file names to Thrift targets", level=LogLevel.DEBUG)
+async def map_thrift_files(protobuf_targets: AllThriftTargets) -> ThriftMapping:
+    stripped_sources_per_target = await MultiGet(
+        Get(StrippedSourceFileNames, SourcesPathsRequest(tgt[ThriftSourceField]))
+        for tgt in protobuf_targets
+    )
```

Comment:
This can be simplified post-https://github.com/pantsbuild/pants/pull/13981. Would it help for me to send a diff?

Reply:
Sure.

Reply:
```diff
diff --git a/src/python/pants/backend/codegen/thrift/dependency_inference.py b/src/python/pants/backend/codegen/thrift/dependency_inference.py
index 1db57c67e..2fc8f0878 100644
--- a/src/python/pants/backend/codegen/thrift/dependency_inference.py
+++ b/src/python/pants/backend/codegen/thrift/dependency_inference.py
@@ -10,7 +10,7 @@ from typing import DefaultDict
 
 from pants.backend.codegen.thrift.subsystem import ThriftSubsystem
 from pants.backend.codegen.thrift.target_types import AllThriftTargets, ThriftSourceField
-from pants.core.util_rules.stripped_source_files import StrippedSourceFileNames
+from pants.core.util_rules.stripped_source_files import StrippedFileName, StrippedFileNameRequest
 from pants.engine.addresses import Address
 from pants.engine.fs import Digest, DigestContents
 from pants.engine.rules import Get, MultiGet, collect_rules, rule
@@ -41,22 +41,20 @@ class ThriftMapping:
 
 @rule(desc="Creating map of Thrift file names to Thrift targets", level=LogLevel.DEBUG)
 async def map_thrift_files(protobuf_targets: AllThriftTargets) -> ThriftMapping:
-    stripped_sources_per_target = await MultiGet(
-        Get(StrippedSourceFileNames, SourcesPathsRequest(tgt[ThriftSourceField]))
+    stripped_file_per_target = await MultiGet(
+        Get(StrippedFileName, StrippedFileNameRequest(tgt[ThriftSourceField].file_path))
         for tgt in protobuf_targets
     )
 
     stripped_files_to_addresses: dict[str, Address] = {}
     stripped_files_with_multiple_owners: DefaultDict[str, set[Address]] = defaultdict(set)
-    for tgt, stripped_sources in zip(protobuf_targets, stripped_sources_per_target):
-        assert len(stripped_sources) == 1
-        stripped_f = stripped_sources[0]
-        if stripped_f in stripped_files_to_addresses:
-            stripped_files_with_multiple_owners[stripped_f].update(
-                {stripped_files_to_addresses[stripped_f], tgt.address}
+    for tgt, stripped_file in zip(protobuf_targets, stripped_file_per_target):
+        if stripped_file.value in stripped_files_to_addresses:
+            stripped_files_with_multiple_owners[stripped_file.value].update(
+                {stripped_files_to_addresses[stripped_file.value], tgt.address}
             )
         else:
-            stripped_files_to_addresses[stripped_f] = tgt.address
+            stripped_files_to_addresses[stripped_file.value] = tgt.address
 
     # Remove files with ambiguous owners.
     for ambiguous_stripped_f in stripped_files_with_multiple_owners:
```